### ISSUE TITLE
[Feature] Ability to manually import an xml feed of articles into a subscribed feed

### ISSUE THREAD

**Issue by austinliuigi** [FreshRSS/FreshRSS#7079]
**Is your feature request related to a problem? Please describe.**

It would be nice to be able to manually import some articles into a feed in the case that they haven't been saved by your instance of FreshRSS for whatever reason, e.g. you subscribed to the feed recently but you want past articles that are no longer advertised on the live feed.

**Describe the solution you’d like**

An option to add a one-time feed of article(s) that you want to manually import into a subscribed feed, so that FreshRSS will save those articles under that subscribed feed. Following the example above about past articles that were missed, this feature would allow you to add an archived version of the subscribed feed from the wayback machine and import the articles that are contained in it.

**Describe alternatives you’ve considered**

Self-hosting a custom rss feed that scrapes for a historically exhaustive list of articles.

**Comment by Alkarex** [comment]
Hello,
In what format would that list of articles be?
If it is the same than our XML export format, then we already support what you ask.
If not, a related feature we have on the roadmap (there is an issue somewhere) is to manually save an article by providing its link.

**Comment by austinliuigi** [comment]
It would be a regular rss feed format.

> If it is the same than our XML export format, then we already support what you ask. 

Would you be referring to exporting your feeds as OPML? Or does FreshRSS allow you to export articles themselves in an XML format? If you’re referring to OPML, I was under the impression that this simply exports references to the feeds you’ve subscribed to but not the articles that your instance of FreshRSS has saved. Please correct me if I’m wrong.

> If not, a related feature we have on the roadmap (there is an issue somewhere) is to manually save an article by providing its link.

Would this allow you to save the article under a certain feed? If so, that would be nice.

**Comment by Alkarex** [comment]
> does FreshRSS allow you to export articles themselves in an XML format?

Yes. Check the export page. Favourite and labelled articles can be exported (and imported).

Edit: sorry, not XML, but JSON

**Comment by Alkarex** [comment]
Can you explain some more what the scenario would be? You would have manually saved an RSS feed in the past, which you want to manually import? That sounds quite a rare need. Do you have such a need at the moment? We can maybe try to find a quick fix for your specific situation.

**Comment by Frenzie** [comment]
I don't know if it's that rare. Feeds or articles on feeds disappear plenty. :-)

But @austinliuigi what you describe in the OP you can already do by editing the URL. Go to manage, change the feed URL to the other feed, clear cache, update, and change it back.

**Comment by austinliuigi** [comment]
> Can you explain some more what the scenario would be? You would have manually saved an RSS feed in the past, which you want to manually import?

You could have manually saved an RSS feed previously, or it could have been archived in something like the Wayback Machine. You could even manually scrape the website to create an ad-hoc feed of all past articles. Importing this would allow you to save those articles, even if you added the feed after those articles stopped being advertised.

> But @austinliuigi what you describe in the OP you can already do by editing the URL. Go to manage, change the feed URL to the other feed, clear cache, update, and change it back.

Oh that's wonderful, thank you. Could you explain why manually clearing the cache is necessary in this case? Is it so that you wouldn't need to wait 15 minutes (or whatever is set for the `“Do not automatically refresh more often than”` setting) to be able to update?

I think being able to directly import the feed would be slightly more convenient so you wouldn't need to temporarily host it and change the feed url twice, but it might not be worth the hassle of implementing it since this method is functionally equivalent, even if less ergonomic. What do you guys think?

**Comment by Alkarex** [comment]
Indeed, if the feeds archives are online like https://web.archive.org/web/20130921183515/http://www.w3.org/blog/feed/ then @Frenzie's suggestion seems sufficient to me.

**Comment by Frenzie** [comment]
> Could you explain why manually clearing the cache is necessary in this case?

I'm not sure how it interacts with changing the URL. Maybe it's not necessary, but I figured I'd include it in case it was.

**Comment by Alkarex** [comment]
The cache would be different when changing the URL

**Comment by math-GH** [comment]
> I think being able to directly import the feed would be slightly more convenient so you wouldn't need to temporarily host it and change the feed url twice, but it might not be worth the hassle of implementing it since this method is functionally equivalent, even if less ergonomic. What do you guys think?

I think it is a simple and good workaround for a very very rare use case
